webapp/src/js/actions/reports.js

```diff
@@ -29,20 +36,117 @@ angular.module('inboxServices').factory('ReportsActions',
         dispatch(ActionUtils.createSingleValueAction(actionTypes.SET_SELECTED_REPORTS, 'selected', selected));
       }
 
+      function setVerifyingReport(verifyingReport) {
+        dispatch(ActionUtils.createSingleValueAction(actionTypes.SET_VERIFYING_REPORT, 'verifyingReport', verifyingReport));
+      }
+
       function updateSelectedReportItem(id, selected) {
         dispatch({
           type: actionTypes.UPDATE_SELECTED_REPORT_ITEM,
           payload: { id, selected }
         });
       }
 
+      function setTitle(model) {
+        dispatch(function(dispatch, getState) {
+          const formInternalId = model.formInternalId || model.form;
+          const forms = Selectors.getForms(getState());
+          const form = _.findWhere(forms, { code: formInternalId });
+          const name = (form && form.name) || (form && form.title) || model.form;
+          globalActions.setTitle(name);
+        });
+      }
+
+      function setRightActionBar() {
+        dispatch(function(dispatch, getState) {
+          const selectMode = Selectors.getSelectMode(getState());
+          const selectedReportsDocs = Selectors.getSelectedReportsDocs(getState());
+          const model = {};
+          const doc =
+            !selectMode &&
+            selectedReportsDocs &&
+            selectedReportsDocs.length === 1 &&
+            selectedReportsDocs[0];
+          if (!doc) {
+            return globalActions.setRightActionBar(model);
+          }
+          model.verified = doc.verified;
+          model.type = doc.content_type;
+          const verifyingReport = Selectors.getVerifyingReport(getState());
+          model.verifyingReport = verifyingReport;
+          if (!doc.contact || !doc.contact._id) {
+            return globalActions.setRightActionBar(model);
+          }
+
+          DB()
+            .get(doc.contact._id)
+            .then(function(contact) {
+              model.sendTo = contact;
+              globalActions.setRightActionBar(model);
+            })
+            .catch(function(err) {
+              globalActions.setRightActionBar(model);
```

Comment:
This weirds me out, I feel like a comment here would be useful (why would we expect it to crash but in a way that is valid enough to set stuff anyway).

Wish promises had a `finally`...

Reply:
It actually does have finally now, just not in the browsers we support: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

This is really ugly anyway because it's throwing the `err` but because the promise isn't returned it can't be caught and handled. I've modified it so it just logs it out and carries on, which means the `then` can be below the `catch` so no `finally` needed!